packages/appium/lib/cli/extension-command.js

```diff
@@ -225,6 +240,29 @@ class ExtensionCliCommand {
     return listData;
   }
 
+  /**
+   * Checks whether the given extension is compatible with the currently installed server
+   *
+   * @param {InstallViaNpmArgs} installViaNpmOpts
+   * @returns {Promise<void>}
+   */
+  async _checkInstallCompatibility({installSpec, pkgName, installType}) {
+    if (INSTALL_TYPE_NPM !== installType) {
+      return;
+    }
+
+    await spinWith(this.isJsonOutput, `Checking if '${pkgName}' is compatible`, async () => {
+      const [serverVersion, extVersionRequirement] = await getRemoteExtensionVersionReq(installSpec);
+      if (serverVersion && extVersionRequirement && !semver.satisfies(serverVersion, extVersionRequirement)) {
+        throw this._createFatalError(
+          `'${installSpec}' cannot be installed because the server version it requires (${extVersionRequirement}) ` +
+          `does not meet the currently installed one (${serverVersion}). Please install ` +
+          `a compatible server version first.`
```

Comment:
```suggestion
          `does not meet the currently installed one (${serverVersion}). Please install ` +
          `a compatible server version, or use a different version of '${pkgName}'.`
```

Reply:
> or use a different version of '${pkgName}'

This message is also shown on upgrade, so I was not sure if it wouldn't be confusing

Reply:
also, it is not clear if there is `a different version` of the package, which would be compatible to the current server version